python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -156,3 +154,41 @@ def compile(model: nn.Module,
                 return pl_model
         else:
             return pl_model
+
+    def quantize(self, model, calib_dataloader, val_dataloader=None, metric: str = None,
+                 backend='inc', conf=None, framework='pytorch_fx', approach='ptsq',
+                 strategy='bayesian', accuracy_criterion=None, timeout=0, max_trials=1):
+        if backend == 'inc':
+            from bigdl.nano.quantization import QuantizationINC
+            quantizer = QuantizationINC(framework=framework, conf=conf, approach=approach,
+                                        strategy=strategy, accuracy_criterion=accuracy_criterion,
+                                        timeout=timeout, max_trials=max_trials)
+            q_litmodel = copy.deepcopy(model)
+            quantizer.model = q_litmodel.model
+            q_approach = quantizer.cfg.quantization.approach
+
+            def eval_func(model_to_eval):
+                if val_dataloader:
+                    q_litmodel.model = model_to_eval
+                    val_outputs = self.validate(q_litmodel, val_dataloader)
+                    return val_outputs[0][f'val/{metric}']
+                else:
+                    return 0
+
+            quantizer.eval_func = eval_func
+
+            if q_approach == 'quant_aware_training':
+                def q_func(model_to_train):
+                    q_litmodel.model = model_to_train
+                    self.fit(q_litmodel, train_dataloaders=calib_dataloader)
```

Comment:
Why do you use nano in training?

Reply:
So I don't need to implement another `train_func`. And we can also benefit from nano acceleration when doing QAT. This part can be removed if now we just need PTQ.

Reply:
Sorry the question is "how do you use nano in training?"

Reply:
Before runing python script, do:
```shell
source bigdl-nano-init
```
or
```shell
bigdl-nano-init python main.py ...
```
And just do everything as normal. Like:
```python
trainer = Trainer(use_ipex=True, ...)
```

Reply:
I don't understand - how does the user perform QAT in this case? Can you give an example?

Reply:
```python
trainer.quantize(model, train_loader, ..., approach='qat')
```
Inside `trainer.quantize(..)`, `trainer.fit(..)` will be called to do QAT in this case.
This is one implementation for QAT. Sorry I will remove QAT in this PR and focus on post-training.
Finally I will implement another PL callback to do QAT as:
```python
trainer = Trainer(callback=[INCQuantAwareTraining(...)])
trainer.fit(train_loader, ...)
```